x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/select_inference_id.tsx

```diff
@@ -135,14 +140,34 @@ export const SelectInferenceId = ({
         setIsInferenceFlyoutVisible(!isInferenceFlyoutVisible);
         setIsCreateInferenceApiLoading(false);
         setInferenceAddError(undefined);
+
+        const trainedModelId =
+          'model_id' in modelConfig.service_settings &&
```

Comment:
nitpick: `modelConfig.service_settings.model_id === Service.elser || modelConfig.service_settings.model_id === Service.elasticsearch` will do the same thing, no need to prepend an exists check.